src/engine/Account-Imap.cpp

```diff
@@ -0,0 +1,1186 @@
+/********************************************************************\
+ * Account-Imap.c -- Account Import Mapping                         *
+ * Copyright (C) 1997 Robin D. Clark                                *
```

Comment:
Don't forget to change the file name in the comments. Depending on how extensive the rewrite you might also consider removing the earlier copyright notices.